### ISSUE TITLE
When onnxruntime_PREFER_SYSTEM_LIB is set, the onnxruntime build now breaks when some packages are missing

### ISSUE THREAD

**Issue by sstamenova** [microsoft/onnxruntime#9403]
**Describe the bug**
This is recent as of: https://github.com/microsoft/onnxruntime/pull/9181. We were using an older version of onnxruntime with onnxruntime_PREFER_SYSTEM_LIB set without issue, but trying to update to the latest version fails because when a package is not found, aliasing does not work.

For example, 
```
if(onnxruntime_PREFER_SYSTEM_LIB)
  find_package(Flatbuffers)
  add_library(flatbuffers ALIAS flatbuffers::flatbuffers)
endif()
if(NOT TARGET flatbuffers::flatbuffers)
  add_subdirectory(external/flatbuffers EXCLUDE_FROM_ALL)
endif()
```

When Flatbuffers is not found, the call to `add_library` will fail because this target does not exist and therefore, cannot be aliased. Instead, the `add_library` call should be guarded by `if(TARGET)` or `if(Flatbuffers_FOUND)`. The same is true in a couple of other locations in the file all stemming from the same commit.

**Urgency**
None

**System information**

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows or Ubuntu 20.04
- ONNX Runtime installed from (source or binary): source
- ONNX Runtime version: commit da56f01
- Python version: 3.9
- Visual Studio version (if applicable): Visual Studio 2019 Developer Command Prompt v16.11.4 (if on Windows)
- GCC/Compiler version (if compiling from source): MSVC 19.29.30136.0 or Clang 11.0.0

**To Reproduce**
- Describe steps/code to reproduce the behavior.
Build onnxruntime on Windows or Ubuntu with `onnxruntime_PREFER_SYSTEM_LIB` set without installing a package for Flatbuffers.

**Expected behavior**
If the package doesn't exist, Flatbuffers is rebuilt and the build succeeds.

**Screenshots**
```
CMake Warning at third_party/onnxruntime/cmake/CMakeLists.txt:1427 (find_package):
  By not providing "FindFlatbuffers.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "Flatbuffers", but CMake did not find one.

  Could not find a package configuration file provided by "Flatbuffers" with
  any of the following names:

    FlatbuffersConfig.cmake
    flatbuffers-config.cmake

  Add the installation prefix of "Flatbuffers" to CMAKE_PREFIX_PATH or set
  "Flatbuffers_DIR" to a directory containing one of the above files.  If
  "Flatbuffers" provides a separate development package or SDK, be sure it
  has been installed.


CMake Error at third_party/onnxruntime/cmake/CMakeLists.txt:1428 (add_library):
  add_library cannot create ALIAS target "flatbuffers" because target
  "flatbuffers::flatbuffers" does not already exist.
```

**Comment by sstamenova** [comment]
As an example of how this can be done better, please see how protobuf is handled:

https://github.com/microsoft/onnxruntime/blob/2f1204a5d5d1d017508c0ec1be47755901458b0c/cmake/CMakeLists.txt#L744

Note that this also needs to be improved since the `if` statement is redundant.

**Comment by skottmckay** [comment]
Resolved by #9404
